Add PDC downloads from server shared with CRDC BT-381

kshakir opened the pull request.

kshakir committed:
Add PDC downloads from server shared with CRDC

breilly2 commented on martha/martha_v3.js:
```diff
@@ -391,7 +408,7 @@ function determineDrsType(url) {
             AUTH_SKIPPED,
             BOND_PROVIDER_DCF_FENCE,
-            ACCESS_METHOD_TYPE_NONE,
+            [ACCESS_METHOD_TYPE_GCS, ACCESS_METHOD_TYPE_S3],
```
Do we want only S3 for now so we don't turn on signed URLs for GCS objects?

kshakir replied:
Discussed. Turns out CRDC hosts data on both and we want to retrieve GCS data if available. Meanwhile, changed BDC's `.accessMethodTypes` to `[ACCESS_METHOD_TYPE_GCS]`

breilly2 commented on martha/martha_v3.js:
```diff
@@ -480,10 +497,10 @@ async function retrieveFromServers(params) {
     console.log(
         `DRS URI '${url}' will use auth required '${sendAuth}', bond provider '${bondProvider}', ` +
-        `and access method type '${accessMethodType}'`
+        `and access method types '${accessMethodTypes.join("', '")}'`
```
Nit: unless you really want the space after the comma, you can use `${accessMethodTypes.toString()}`.

breilly2 commented on test/martha/martha_v3.test.js:
```diff
@@ -555,6 +557,58 @@ test.serial('martha_v3 parses a Gen3 CRDC CIB URI response correctly', async (t)
+    const {
+        access_methods: { 0: { access_id: accessId, access_url: { url: s3Url } } },
+    } = pdcResponse;
+    const objectId = drsUri.replace(/.*:/, '');
```
Might be more clear to build `drsUri` from `objectId`:
```
const objectId = '206dfaa6-bcf1-4bc9-b2d0-77179f0f48fc'
const drsUri = `drs://dg.4DFC:${objectId}`
```

kshakir committed:
PR Fixup:
- Switched BDC's .accessMethodTypes
- Changed log formatting of accessMethodTypes
- Use test objectId to create drsUri

mcovarr commented on martha/martha_v3.js:
```diff
@@ -369,7 +397,7 @@ function determineDrsType(url) {
             AUTH_SKIPPED,
             BOND_PROVIDER_ANVIL,
-            ACCESS_METHOD_TYPE_NONE,
+            ACCESS_METHOD_TYPES_NONE,
```
TOL would `[ACCESS_METHOD_TYPE_GCS]` be more correct here too?

kshakir committed:
PR Fixup: Match The AnVIL .accessMethodTypes with BDC

breilly2 reviewed (approved):
👍   I've been waiting on non-prod PDC test data, but still haven't gotten any. Ultimately, I ran locally with a tweak to the preference order for CRDC to be S3, GS. I used the CRDC object used in our tests (since that has both GS and S3) and successfully retrieved a signed URL.

breilly2 commented:
Michael Baumann managed to mine some "real" (staging) PDC data. Example: drs://dg.4DFC:dg.4DFC/f2ffba75-5197-11e9-9a07-0a80fada099c. I tested this branch against this and successfully retrieved a signed URL and downloaded the resulting "fasta" file (small, looks to be truncated or synthetic).

kshakir committed:
PR Fixup: Use PDC CIB URI from notebook

kshakir committed:
PR Fixup: now with working tests!

kshakir committed:
PR Fixup: more dependency updates
